plugins/extractors/bigquery/bigquery.go

```diff
@@ -137,11 +142,20 @@ func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error)
 
 // Create big query client
 func (e *Extractor) createClient(ctx context.Context) (*bigquery.Client, error) {
-	if e.config.ServiceAccountJSON == "" {
+	if e.config.ServiceAccountBase64 == "" && e.config.ServiceAccountJSON == "" {
 		e.logger.Info("credentials are not specified, creating bigquery client using default credentials...")
 		return bigquery.NewClient(ctx, e.config.ProjectID)
 	}
 
+	if e.config.ServiceAccountBase64 != "" {
+		serviceAccountJSON, err := base64.StdEncoding.DecodeString(e.config.ServiceAccountBase64)
+		if err != nil {
+			e.logger.Fatal(fmt.Sprintf("failed to decode base64 service account, err : %v\n", err))
```

Comment:
Why not just return the error from here? That way, we can test it as well.

Reply:
modified it to return error and added a test case for invalid config.